### PR TITLE
Add EditorConfig file

### PR DESCRIPTION
**Pull request by AlexWayfer**
More info here: https://editorconfig.org/

Also fix some line-end spaces, file-end newlines.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
```
Comment (thomasrockhu) [inline .editorconfig:9]: Can these be alphabetized?

**Review by thomasrockhu — CHANGES_REQUESTED** [review]
Will approve pending nit

File: .editorconfig (hunk 1 above)
Comment (thomasrockhu) [inline .editorconfig:9]:
```suggestion
charset = utf-8
end_of_line = lf
indent_size = 2
indent_style = space
insert_final_newline = true
trim_trailing_whitespace = true
```

Comment (AlexWayfer) [reply .editorconfig]: Of course, if you want to. Suggested code has been accepted.

**Review by thomasrockhu — APPROVED** [review]

**Pull request merged**